### PR TITLE
Create action

### PR DESCRIPTION
**Pull request by quesabe**

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        ref: ${{ inputs.commit-hash }}
+    - uses: ottofeller/npm-run@main
+      with:
+        node-version: ${{ inputs.node-version }}
+        registry-url: ${{ inputs.registry-url }}
+        scope: ${{ inputs.scope }}
+        command: npm publish
```
Comment (gvidon) [inline action.yml:26]: @quesabe let's allow for setting `--access` value from the input params.

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Publish npm package
+description: 'A GitHub action publishing an npm package'
```
Comment (gvidon) [inline action.yml:2]: @quesabe I would stick to using double quotes.

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+inputs:
+  commit-hash:
```
Comment (gvidon) [inline action.yml:4]: @quesabe can we make the commit hash as optional param and by default publish from the primary branch?

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    description: 'A commit hash that will be used as a reference for package publishing'
+  node-version:
```
Comment (gvidon) [inline action.yml:6]: @quesabe let's set 16 as a default value.

File: action.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    description: 'Version Spec of the version to use.  Examples: 12.x, 10.15.1, >=10.15.0'
+  registry-url:
```
Comment (gvidon) [inline action.yml:8]: @quesabe are these params really optional? Won't action fail if it doesn't find them to be set?

**Review by gvidon — CHANGES_REQUESTED** [review]

File: action.yml (hunk 5 above)
Comment (quesabe) [reply action.yml:8]: Yes, they are. The optionality is described in node-setup readme. As far as I remember, node would default to 12. registry-url is written to the `.npmrc` file and if not supplied nom will use default value of `https://registry.npmjs.org`.

File: action.yml (hunk 2 above)
Comment (quesabe) [reply action.yml]: Removed most of single quotes, the remaining updated to double.

File: action.yml (hunk 3 above)
Comment (quesabe) [reply action.yml]: It is already optional per actions/checkout docs. Updated descriptions.

**Review by gvidon — APPROVED** [review]

**Pull request merged**